docs/querying/functions.md

```diff
@@ -699,68 +721,76 @@ observation constitutes a counter reset, but also the disappearance of any
 previously populated bucket, an increase in bucket resolution, or a decrease of
 the zero-bucket width.
 
-`resets` should only be used with counters and counter-like native
-histograms.
+`resets` should only be used with counters (for both floats and histograms).
 
-If the range vector contains a mix of float and histogram samples for the same
-series, counter resets are detected separately and their numbers added up. The
-change from a float to a histogram sample is _not_ considered a counter
-reset. Each float sample is compared to the next float sample, and each
-histogram is comprared to the next histogram.
+A float sample followed by a histogram sample, or vice versa, counts as a
+reset. A counter histogram sample followed by a gauge histogram sample, or vice
+versa, also counts as a reset (but note that `resets` should not be used on
+gauges in the first place, see above).
```

Comment:
Can you clarify why the transition between exponential and custom bucket histograms is not treated as a reset?

Reply:
It should be treated as a counter reset (and hopefully is). I have forgotten to mention schema changes above. I'll add this and also verify that all incompatible schema changes are counted as resets.